src/main/java/net/sigmarik/abilitymod/command/TraitCommand.java

```diff
@@ -53,16 +74,43 @@ private static int addTrait(CommandContext<ServerCommandSource> context) throws
     }
 
     private static int removeTrait(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
-        String trait = StringArgumentType.getString(context, "trait");
+        String[] traits = StringArgumentType.getString(context, "trait").split(" ");
         java.util.Collection<net.minecraft.server.network.ServerPlayerEntity> players =
                 EntityArgumentType.getPlayers(context, "players");
 
         ServerState states = ServerState.getTraitStates(context.getSource().getServer());
 
+        boolean allTraits = false;
+        for (String trait : traits) {
+            allTraits |= trait.equals("*");
+        }
+
+        List<String> traitCollection = List.of(traits);
+
         for (ServerPlayerEntity player : players) {
-            AbilityMod.LOGGER.info("Removing trait " + trait + " from player " + player.getName().getString());
+            if (allTraits) {
+                traitCollection = states.getTraitList(player.getUuid()).stream().toList();
+            }
 
-            states.removeTrait(player.getUuid(), trait);
+            for (String trait : traitCollection) {
+                AbilityMod.LOGGER.info("Removing trait " + trait + " from player " + player.getName().getString());
+                states.removeTrait(player.getUuid(), trait);
+
+                if (trait.equals(AbilityMod.TRAIT_FAST)) {
+                    EntityAttributeInstance speed = player.getAttributes().getCustomInstance(EntityAttributes.GENERIC_MOVEMENT_SPEED);
+                    EntityAttributeModifier speedModifier = new EntityAttributeModifier(AbilityMod.FAST_MODIFIER_UUID, "Fast", AbilityMod.FAST_COEFFICIENT, EntityAttributeModifier.Operation.MULTIPLY_BASE);
```

Comment:
Wow, I really should have got a wide monitor for this review...

Anyway, there is another LONG BOIIIIIIIIII